23 informers workload

squeedee opened the pull request.

Commit:
Workload informers

[Updates #23]

Co-authored-by: Emily Johnson <emjohnson@vmware.com>

emmjohnson commented on tests/integration/supplychain/workload_reconciler_test.go:
```diff
@@ -274,4 +278,164 @@ var _ = Describe("WorkloadReconciler", func() {
+				return test.Status.Conditions, err
+			}).Should(BeNil())
+
+			_ = c.Get(ctx, client.ObjectKey{Name: "test-resource", Namespace: testNS}, test)
```
Remove this

martyspiewak commented on tests/kuttl/supplychain/templates-ytt/01-assert.yaml:
```diff
@@ -0,0 +1,27 @@
+#      reason: Ready
+#    - type: Ready
+#      status: "True"
+#      reason: Ready
```
Is there a reason this file is commented out?

emmjohnson replied:
no

martyspiewak commented on pkg/registrar/map_functions.go:
```diff
@@ -40,6 +40,53 @@ type Mapper struct {
+
+	var supplyChains []*v1alpha1.ClusterSupplyChain
+	for scIndex := range list.Items {
+		sc := list.Items[scIndex]
```
```suggestion
	for _, sc := range list.Items {
```
nitpick- this would be more consistent with what we do in `ClusterDeliveryToDeliverableRequests` and `ClusterSupplyChainToWorkloadRequests` below and is a bit easier to reason about while reading through the code.

emmjohnson replied:
That actually causes major problems. See https://play.golang.org/p/GzU80LswSEt

squeedee replied:
we also inlined a comment to guide people.

martyspiewak commented on tests/integration/supplychain/policy.yaml:
```diff
@@ -15,11 +15,13 @@
 apiVersion: audit.k8s.io/v1 # This is required.
 kind: Policy
 # Don't generate audit events for all requests in RequestReceived stage.
```
```suggestion

```
That comment seems to be irrelevant now as it was on the `omitStages` portion that was removed.

martyspiewak commented:
Looks great! Left a few comments with some small changes but other than that lgtm!

Commit:
Address reviewer's comments

- Explain range reusing the same value
- clean up audit policy
- reinstate missing code

Co-authored-by: Emily Johnson <emjohnson@vmware.com>

Commit:
Do not use pointers for map and reduce

Co-authored-by: Rasheed Abdul-Aziz <rabdulaziz@vmware.com>
